### PR TITLE
Fix slower CV_PAUSE on SkyLake and above.

### PR DESCRIPTION
**Pull request by vrabaud**
This is fixing https://github.com/opencv/opencv/issues/22852

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work

### CODE REVIEW

**Comment by asmorkalov** [comment]
```
/home/ci/opencv/modules/core/src/parallel_impl.cpp:419:85: error: '__rdtsc' was not declared in this scope
                     CV_PAUSE(16);
                                                                                     ^
/home/ci/opencv/modules/core/src/parallel_impl.cpp: In member function 'void cv::ThreadPool::run(const cv::Range&, const cv::ParallelLoopBody&, double)':
/home/ci/opencv/modules/core/src/parallel_impl.cpp:701:97: error: '__rdtsc' was not declared in this scope
                                 CV_PAUSE(16);
                                                                                                 ^
make[2]: *** [modules/core/CMakeFiles/opencv_core.dir/src/parallel_impl.cpp.o] Error 1
```

File: modules/core/src/parallel_impl.cpp [hunk 1]
```diff
@@ -52,11 +52,14 @@ DECLARE_CV_PAUSE
...
 # if defined __GNUC__ && (defined __i386__ || defined __x86_64__)
+#   include <x86intrin.h> /* for __rdtsc */
 #   if !defined(__SSE2__)
       static inline void cv_non_sse_mm_pause() { __asm__ __volatile__ ("rep; nop"); }
 #     define _mm_pause cv_non_sse_mm_pause
 #   endif
-#   define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { _mm_pause(); } } while (0)
+// 5 * v is meants for backward compatibility: with pre-Skylake CPUs, _mm_pause took 4 or 5 cycles.
+// With post-Skylake CPUs, _mm_pause takes 140 cycles.
+#   define CV_PAUSE(v) do { const uint64_t __delay = 5 * v; uint64_t __init = __rdtsc(); do { _mm_pause(); } while ((__rdtsc() - __init) < __delay); } while (0)
```
[the remark this answers is not in the document]
Comment (vrabaud) [reply modules/core/src/parallel_impl.cpp:62]: We could indeed use _mm_pause once, that would save some cycles (I thought RDTSC was 18 cycles) BUT that would not be compatible with older CPUs where ```_mm_pause``` is only 4/5 cycles.
Do we have a way to check for SkyLake and above CPUs ? I'll gladly make a patch for that.

[the remark this answers is not in the document]
Comment (vrabaud) [reply modules/core/src/parallel_impl.cpp:62]: Indeed, so what do you suggest: that I make a new pull request that only does one ```_mm_pause``` where defined ?
And for older CPUs, ```OPENCV_THREAD_POOL_ACTIVE_WAIT_PAUSE_LIMIT``` should be used ? I am totally fine with that.

[the remark this answers is not in the document]
Comment (vrabaud) [reply modules/core/src/parallel_impl.cpp:62]: Done in https://github.com/opencv/opencv/pull/22988

**Pull request merged**